File: src/client/xnative/lion.xdroid/Popup/NumberPopupActivity.cs
using Android.App;
using Android.Content.PM;
using Android.OS;
using Android.Views;
using Android.Widget;
using Lion.XDroid;
using Lion.XDroid.Control;
using Lion.XDroid.Libs;
using System.Collections.Generic;

namespace Lion.XDroid.Popup
{
    [Activity(Label = "NumberPopupActivity", Theme = "@android:style/Theme.Translucent.NoTitleBar", ScreenOrientation = ScreenOrientation.Portrait)]
    public class NumberPopupActivity : Activity
    {
        int selected_number;

        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            SetContentView(Resource.Layout.activity_number_popup);

            this.InitData();
            this.InitEvent();
            this.InitSize();
        }

        private void InitSize()
        {
            var btn_ok = FindViewById<Button>(Resource.Id.btn_ok);
            var btn_cancel = FindViewById<Button>(Resource.Id.btn_cancel);

            btn_ok.LayoutParameters = AppCommon.SNG.ResizeDP(this, btn_ok.LayoutParameters);
            btn_cancel.LayoutParameters = AppCommon.SNG.ResizeDP(this, btn_cancel.LayoutParameters);
        }

        private void InitEvent()
        {
            var btn_ok = FindViewById<Button>(Resource.Id.btn_ok);
            btn_ok.Click += ListenerClick;
            btn_ok.Touch += ListenerTouch;

            var btn_cancel = FindViewById<Button>(Resource.Id.btn_cancel);
            btn_cancel.Click += ListenerClick;
            btn_cancel.Touch += ListenerTouch;
        }

        private void ListenerTouch(object sender, Android.Views.View.TouchEventArgs e)
        {
            var _view = (Button)sender;
            if (e.Event.Action == MotionEventActions.Down)
            {
                _view.Background.SetColorFilter(0x10000000);
                _view.Invalidate();
            }
            else if (e.Event.Action == MotionEventActions.Up)
            {
                _view.Background.SetColorFilter(0x00000000);
                _view.Invalidate();
            }
            else if (e.Event.Action == MotionEventActions.Cancel)
            {
                _view.Background.SetColorFilter(0x00000000);
                _view.Invalidate();
            }

            e.Handled = false;
        }

        private void ListenerClick(object sender, System.EventArgs e)
        {
            var _view = (Button)sender;
            switch (_view.Id)
            {
                case Resource.Id.btn_ok:
                    this.Intent.PutExtra("number", selected_number);
                    SetResult(Result.Ok, this.Intent);
                    Finish();
                    break;
                case Resource.Id.btn_cancel:
                    SetResult(Result.Canceled, this.Intent);
                    Finish();
                    break;
            }
        }

        private void InitData()
        {
            var lv_popup = FindViewById<ListView>(Resource.Id.lv_popup);
            var min = this.Intent.GetIntExtra("min", 1);
            var max = this.Intent.GetIntExtra("max", 10);
            var reverse = this.Intent.GetBooleanExtra("reverse", true);

            var _number_list = new List<int>();
            if (reverse)
            {
                this.selected_number = max;
                for (int i = max; i >= min; i--)
                    _number_list.Add(i);
            }
            else
            {
                this.selected_number = min;
                for (int i = min; i <= max; i++)
                    _number_list.Add(i);
            }

            var adapter = new ListViewResultAdapter(this, Resource.Layout.listview_result_custom, _number_list);
            adapter.ChangePosition += (s, e) =>
            {
                this.selected_number = e.Position;
            };

            lv_popup.Adapter = adapter;
        }
    }
}

File: src/client/xcode/Lion.Droid/app/src/main/java/com/odinsoftware/lottolion/popup/NumberPopupActivity.java
package com.odinsoftware.lottolion.popup;

import android.app.Activity;
import android.content.Intent;
import android.graphics.PorterDuff;
import android.media.Image;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.MotionEvent;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.ListView;
import android.widget.NumberPicker;

import com.odinsoftware.lottolion.R;
import com.odinsoftware.lottolion.control.ListViewResultAdapter;
import com.odinsoftware.lottolion.service.LLCommon;

import java.util.ArrayList;
import java.util.List;

public class NumberPopupActivity extends Activity {

    //NumberPicker np_popup;
    Intent intent;
    int result;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_number_popup);

        InitData();
        InitEvent();
        InitSize();
    }

    private void InitSize() {
        Button btn_ok = (Button)findViewById(R.id.btn_ok);
        Button btn_cancel = (Button)findViewById(R.id.btn_cancel);

        btn_ok.setLayoutParams(LLCommon.Get().ResizeDP(this,btn_ok.getLayoutParams()));
        btn_cancel.setLayoutParams(LLCommon.Get().ResizeDP(this,btn_cancel.getLayoutParams()));
    }

    private void InitEvent() {
        Button btn_ok = (Button)findViewById(R.id.btn_ok);
        Button btn_cancel = (Button)findViewById(R.id.btn_cancel);

        btn_ok.setOnClickListener(listener_click);
        btn_cancel.setOnClickListener(listener_click);
        btn_ok.setOnTouchListener(listener_touch);
        btn_cancel.setOnTouchListener(listener_touch);

    }

    private void InitData() {
        intent = getIntent();
        ListView lv_popup = (ListView)findViewById(R.id.lv_popup);
        int min =intent.getIntExtra("min",1);
        int max= intent.getIntExtra("max",10);
        boolean reverse = intent.getBooleanExtra("reverse",true);

        ArrayList<Integer> list = new ArrayList<>();
        if(reverse){
            result= max;
            for(int i= max; i>= min ; i--){
                list.add(i);
            }
        }
        else{
            result= min;
            for(int i= min; i<= max ; i++){
                list.add(i);
            }
        }
        ListViewResultAdapter adapter = new ListViewResultAdapter(this,R.layout.listview_result_custom,list);
        adapter.resultClickListener = new ListViewResultAdapter.ResultOnClickListener() {
            @Override
            public void onClickListener(int position) {
                result = position;
            }
        };
        lv_popup.setAdapter(adapter);

    }
    View.OnClickListener listener_click = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            switch (v.getId()){
                case R.id.btn_ok:
                    intent.putExtra("number",result);
                    setResult(RESULT_OK, intent);
                    finish();
                    break;
                case R.id.btn_cancel:
                    setResult(RESULT_CANCELED, intent);
                    finish();
                    break;
            }
        }
    };
    View.OnTouchListener listener_touch =  new View.OnTouchListener() {
        @Override
        public boolean onTouch(View v, MotionEvent event) {
            Button view = (Button)v;
            if(event.getAction() == MotionEvent.ACTION_DOWN){
                view.getBackground().setColorFilter(0x10000000, PorterDuff.Mode.SRC_OVER);
                view.invalidate();
            }
            else if(event.getAction() == MotionEvent.ACTION_UP){
                view.getBackground().setColorFilter(0x00000000, PorterDuff.Mode.SRC_OVER);
                view.invalidate();
            }
            else if(event.getAction() == MotionEvent.ACTION_CANCEL){
                view.getBackground().setColorFilter(0x00000000, PorterDuff.Mode.SRC_OVER);
                view.invalidate();
            }
            return false;
        }
    };
}
